core/primitives/src/runtime/in_memory_contract.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::HashMap;
+use std::sync::RwLock;
+use std::sync::Arc;
+use crate::{
+    hash::CryptoHash,
+    types::AccountId,
+};
+use near_vm_errors::VMError;
+
+pub type InMemoryContracts = Arc<RwLock<HashMap<AccountId, Option<(CryptoHash, Result<wasmer_runtime::Module, VMError>)>>>>;
```

Comment:
Hm, taking a bird's eye view, the fact fact the datastructure for in-memory cache which is used by the vm-runner crate, is defined outside the vm-runner crate feels wrong. 

I guess it's reasonable place to do this, given the current structure of code. But the structure itself is suspicious .

Reply:
this should be struct InMemoryContracts(Arc ...) instead of type. but it should be here because primitivies/src/runtime/apply_state.rs is the place to use it, if it were in runtime crate there would be an additional circular dependency primitives depends on runtime